tests/taskcluster/bin/raptor_graph

```diff
@@ -107,34 +180,61 @@ if (process.argv[2] == '--full') {
 var nextYear = new Date(Date.now());
 nextYear.setFullYear(nextYear.getFullYear() + 1);
 nextYear = nextYear.toJSON();
-
-// Expand all tasks based on the config object
-Object.keys(config).forEach(function(raptorTask) {
-  var definition = config[raptorTask].definition;
-  if (definition == 'raptor_result') {
-    config[raptorTask].params.completedBranchTask = raptorGraph.tasks[0].taskId;
-    config[raptorTask].params.completedPatchTask = raptorGraph.tasks[1].taskId;
-  }
-
-  taskIdMap[raptorTask] = slugid.v4();
-  var task = loadTask(definition, params(config[raptorTask].params, {
-    // Useful for environment variables.
-    process: process,
-    // For dependencies.
-    nextYear: nextYear
-  }));
-  task.taskId = taskIdMap[raptorTask];
-  if (definition == 'raptor_result') {
-    task.requires = [raptorGraph.tasks[0].taskId, raptorGraph.tasks[1].taskId];
-  }
-  raptorGraph.tasks.push(task);
-
-  return;
-});
-
-raptorGraph.tasks = raptorGraph.tasks.map(function(task) {
-  return template(task, { tasks: taskIdMap });
-});
-
-// Output the entire task graph for extension....
-process.stdout.write(JSON.stringify(raptorGraph, null, 2));
+var branchTask, patchTask, reqTask1, reqTask2, branchElement, patchElement;
+
+getEmulator()
+  .then(setpatchTaskRev)
+  .then(function() {
+
+    // Expand all tasks based on the config object
+    Object.keys(config).forEach(function(raptorTask) {
```

Comment:
Is the order of the keys returned sorted consistently in such a way that you'll always have the task IDs for the two launch tasks before processing the raptor_result task?